src/logger/httpLoggerRules_test.go

```diff
@@ -93,13 +95,104 @@ func TestUsesAllowHttpUrlRules(t *testing.T) {
 
 // test uses copy session field rules test
 
-// func TestUsesCopySessionFieldRules(t *testing.T) {
-
-// }
+func TestUsesCopySessionFieldRules(t *testing.T) {
+	// helper for function tests
+	helper := NewTestHelper()
+	// requests used for all tests in function
+	request := NewMockRequestWithJson2()
+	request.Session().SetAttribute("butterfly", "poison") //getting session and setting an attribute of it
+	request.Session().SetAttribute("session_id", "asdf1234")
+	// tests copy all of session field
+	queue := make([]string, 0)
+	logger := NewHttpLoggerQueueRules(queue, "copy_session_field /.*/")
+	httpMessage.Send(logger, request, MockResponseWithHTML(), helper.mockHTML, helper.mockJSON)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"session_field:butterfly\",\"poison\"]"), "queue did not contain expected values")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"session_field:session_id\",\"asdf1234\"]"), "queue did not contain expected values")
+	// tests copy specifically session_id
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "copy_session_field /session_id/")
+	httpMessage.Send(logger, request, MockResponseWithHTML(), helper.mockHTML, helper.mockJSON)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, false, strings.Contains(queue[0], "[\"session_field:butterfly\",\"poison\"]"), "queue contains unexpected value")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"session_field:session_id\",\"asdf1234\"]"), "queue did not contain expected values")
+	// tests copy non matching term
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "copy_session_field /blah/")
+	httpMessage.Send(logger, request, MockResponseWithHTML(), helper.mockHTML, helper.mockJSON)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, false, strings.Contains(queue[0], "[\"session_field:"), "queue contains unexpected value")
```

Comment:
I'm assuming this is supposed to be 0 and not 1 because java's `get()` method starts indexing from 1

Reply:
I'm not sure I see what you're referring to here, It's accessing the queues at 0. but the length of the queue should be 1.